amethyst_ui/src/transform.rs

```diff
@@ -0,0 +1,23 @@
+use specs::{Component, DenseVecStorage};
+
+/// The raw pixels on screen that are populated.
+///
+/// TODO: Eventually this should be either replaced by a citrine type, or citrine may just
+/// populate it.
+pub struct UiTransform {
```

Comment:
Maybe we should limit this to the size and take the position from `Transform`?

Reply:
I feel it's probably important to keep the concepts separate as this is in pixels, whereas Transform uses unspecified units.

Reply:
Okay, true. But if we already have that advantage, we should really do

```rust
pub enum Dimension {
    /// In pixels
    Absolute(f32),
    /// Multiplied with window width / height
    Relative(f32),
}
```

Reply:
Also, I think `Bounds` would be more accurate.

Reply:
I'm thinking `amethyst_ui` might just be glue between `citrine` and our renderer, although we could just merge this into `citrine` instead and move it into the amethyst repo.

Reply:
@torkleyy I was mostly just trying to sidestep the design decisions we talked about on gitter before.  This structure's purpose is to be replaced, it's meant to be slightly sub-optimal.